test/extended/quota/clusterquota.go

```diff
@@ -189,6 +199,74 @@ var _ = g.Describe("[sig-api-machinery][Feature:ClusterResourceQuota]", func() {
 
 				t.Fatalf("unexpected error: %v", err)
 			}
+
+			// test templates are counted correctly
+			template := &templatev1.Template{}
+			template.GenerateName = "test"
+			if _, err := clusterAdminTemplateClient.TemplateV1().Templates(firstProjectName).Create(context.Background(), template, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			if err := waitForQuotaStatus(clusterAdminQuotaClient, cq.Name, func(quota *quotav1.ClusterResourceQuota) error {
+				q := quota.Status.Total.Used["count/templates.template.openshift.io"]
+				if i, ok := q.AsInt64(); ok {
+					if i == 1 {
+						return nil
+					}
+					return fmt.Errorf("%d != 1", i)
+				}
+				return fmt.Errorf("quota=%+v AsInt64() failed", q)
+			}); err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+
+			if _, err := clusterAdminTemplateClient.TemplateV1().Templates(secondProjectName).Create(context.Background(), template, metav1.CreateOptions{}); !apierrors.IsForbidden(err) {
+				framework.Logf("unexpected err during creation: %v", err)
+				list, err := clusterAdminQuotaClient.QuotaV1().AppliedClusterResourceQuotas(secondProjectName).List(context.Background(), metav1.ListOptions{})
+				if err == nil {
+					t.Errorf("quota is %#v", list)
+				}
+
+				list2, err := clusterAdminTemplateClient.TemplateV1().Templates("").List(context.Background(), metav1.ListOptions{})
+				if err == nil {
+					t.Errorf("Templates is %#v", list2)
+				}
+
+				t.Fatalf("unexpected error: %v", err)
+			}
+
+			// test that CRD resources are counted correctly
+			serviceMonitorGVR := schema.GroupVersionResource{Group: "monitoring.coreos.com", Version: "v1", Resource: "servicemonitors"}
```

Comment:
this should start passing once https://github.com/openshift/cluster-policy-controller/pull/115 merges